include/clad/Differentiator/Array.h

```diff
@@ -102,16 +109,16 @@ template <typename T> class array {
   /// Returns the size of the underlying array
   CUDA_HOST_DEVICE std::size_t size() const { return m_size; }
   /// Returns the ptr of the underlying array
-  CUDA_HOST_DEVICE T* ptr() const { return m_arr; }
-  CUDA_HOST_DEVICE T*& ptr_ref() { return m_arr; }
+  CUDA_HOST_DEVICE T* ptr() const PUREFUNC { return m_arr; }
```

Comment:
```suggestion
  CUDA_HOST_DEVICE PUREFUNC T* ptr() const  { return m_arr; }
```

Can we move the attributes to go first?

Reply:
fixed 👍🏼